### PR TITLE
checkIfLocationInUrlParams, added xstate-inspect

### PR DESCRIPTION
**Pull request by thisispatrick33**

### CODE REVIEW

File: middleware/xstate.global.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { interpret } from 'xstate/lib'
 const convertStateToUrl = (state) => {
   return `/${state.toLowerCase()}`
 }
-export default defineNuxtRouteMiddleware((to, from) => {
+export default defineNuxtRouteMiddleware((to, _) => {
```
Comment (peter9ke) [inline middleware/xstate.global.ts:6]: I think you could delete _

File: components/DebugTag.vue [hunk 2]
```diff
@@ -8,7 +8,7 @@
 </template>
 <script setup lang="ts">
 import { computed } from 'vue'
-
+import { useRoute } from '#imports'
```
Comment (peter9ke) [inline components/DebugTag.vue:11]: #imports ? I see this first time

Comment (jakubednar) [reply components/DebugTag.vue:11]: useRoute is globally imported from Nuxt but we decided that we want to import everything manually so #imports is the place from where you can import nuxt functions and other things

File: composables/appointmentState.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { useMachine } from '@xstate/vue'
+import appointmentMachine from '@/machines/appointmentMachine'
+
+export const useAppointmentMachine = () =>
+  useState('appointmentMachine', () => {
+    return useMachine(appointmentMachine, { devTools: true })
```
[the remark this answers is not in the document]
Comment (jakubednar) [reply composables/appointmentState.ts:6]: we tried to implement it but it seems that it does not work properly in current nuxt version

File: components/MobileSearchWithSelectButtons.vue [hunk 4]
```diff
@@ -6,16 +6,36 @@
...
 <script setup lang="ts">
+import { computed } from 'vue'
 import Spacer from '@/components/Spacer.vue'
 import SelectButton from '@/components/SelectButton.vue'
+import { useAppointmentMachine } from '@/composables/appointmentState'
+
+const appointmentMachine = useAppointmentMachine()
+
+const locationSortState = computed(() => {
+  return appointmentMachine.value.state.children['locationSort'].state
```
Comment (peter9ke) [inline components/MobileSearchWithSelectButtons.vue:35]: this state will be always available ?

Comment (peter9ke) [reply components/MobileSearchWithSelectButtons.vue:35]: or we need something like this ```appointmentMachine.value.state.children?.['locationSort']?.state```

File: composables/appointmentState.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { useMachine } from '@xstate/vue'
+import appointmentMachine from '@/machines/appointmentMachine'
+
+export const useAppointmentMachine = () =>
+  useState('appointmentMachine', () => {
+    return useMachine(appointmentMachine, {
+      devTools: true,
```
Comment (peter9ke) [inline composables/appointmentState.ts:7]: we need to handle this by env variable

**Pull request merged**